### PR TITLE
Add Mbed CAN code

### PR DESCRIPTION
**Pull request by annaelisew**

### CODE REVIEW

File: bms/src/BmsThread.h [hunk 1]
```diff
@@ -202,23 +202,17 @@ class BMSThread {
         serial->printf("Current: %d\n", gCurrent);
```
Comment (colescott) [inline bms/src/BmsThread.h:202]: Let's just kill this

File: bms/src/Main.cpp [hunk 2]
```diff
@@ -33,29 +33,15 @@
 //    CANOptions::config<CANOptions::BaudRate::k500k, false>();
```
Comment (colescott) [inline bms/src/Main.cpp:33]: This shouldn't be needed anymore. We can get rid of it

File: bms/src/BmsThread.h [hunk 3]
```diff
@@ -196,29 +197,21 @@ class BMSThread {
...
+      
       // Send CAN
       for (size_t i = 0; i < BMS_BANK_COUNT; i++) {
-        auto txmsg = BMSTempMessage(i, allTemps + (BMS_BANK_TEMP_COUNT * i));
-        canTransmit(&BMS_CAN_DRIVER, CAN_ANY_MAILBOX, &txmsg, TIME_MS2I(100));
+        auto temps = std::array<int8_t, BMS_BANK_TEMP_COUNT>();
+        std::transform(allTemps.begin() + (BMS_BANK_TEMP_COUNT * i),
+                       allTemps.begin() + (BMS_BANK_TEMP_COUNT * (i + 1)),
+                       temps.begin(),
+                       [](tl::optional<int8_t> t) { return t.value_or(-127); });
```
Comment (calcmogul) [inline bms/src/BmsThread.h:208]: A comment should be added that says what this std::transform() is for. I assume it's converting an array of optional bytes to an array of bytes where the invalid bytes are padded as 0xFF.

File: bms/src/Can.h [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  data[3] = maxVoltage;
+  data[4] = minVoltage >> 8;
+  data[5] = minVoltage;
+  data[6] = maxTemp;
+  data[7] = minTemp;
+  return CANMessage(kFuncIdBmsStat, data);
+}
+
+CANMessage BMSVoltageMessage (uint8_t row, uint16_t * voltages) {
+  unsigned char data[8];
```
Comment (calcmogul) [inline bms/src/Can.h:46]: Is there a reason `uint8_t` is used in function parameters but `unsigned char` is used for local variables?

**Comment by colescott** [comment]
The current build error is related to #13

**Comment by colescott** [comment]
Will rebase. stand by

File: bms/src/Can.h (hunk 4 above)
Comment (colescott) [reply bms/src/Can.h:46]: For the record, I agree on preferring stdint. It just makes everything make more sense

File: bms/src/BmsThread.h [hunk 5]
```diff
@@ -45,7 +46,7 @@ class BMSThread {
   }
   void threadWorker() {
     uint16_t* allVoltages = new uint16_t[BMS_BANK_COUNT * BMS_BANK_CELL_COUNT];
-    tl::optional<int8_t>* allTemps = new tl::optional<int8_t>[BMS_BANK_COUNT * BMS_BANK_CELL_COUNT];
+    auto allTemps = std::array<tl::optional<int8_t>, BMS_BANK_COUNT * BMS_BANK_CELL_COUNT>();
```
Comment (calcmogul) [inline bms/src/BmsThread.h:49]: Need a `#include <array>` at the top (include-what-you-use).

File: bms/src/BmsThread.h [hunk 6]
```diff
@@ -11,6 +11,8 @@
 #include "EnergusTempSensor.h"
 #include "LTC6811.h"
 #include "LTC6811Bus.h"
+#include "Can.h"
+#include <array>
```
Comment (calcmogul) [inline bms/src/BmsThread.h:15]: These headers are in the wrong order. See https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes for the rationale.

For this file and platform, the correct order would probably be the following (comments are for explanation; you wouldn't actually write them):
```cpp
// C++ system headers
#include <array>
#include <initializer_list>
#include <vector>

// Thirdparty library headers
#include <mbed.h>
#include <rtos.h>
#include <tl/optional.hpp>

// Local project headers
#include "BmsConfig.h"
#include "Can.h"
#include "EnergusTempSensor.h"
#include "LTC6811.h"
#include "LTC6811Bus.h"
```
assuming

Comment (calcmogul) [reply bms/src/BmsThread.h:15]: Autoformatters exist for this, by the way: https://github.com/wpilibsuite/styleguide/tree/master/wpiformat, or even clang-format itself if you're OK with it getting the ordering wrong sometimes.

File: bms/src/BmsThread.h [hunk 7]
```diff
@@ -1,13 +1,15 @@
 #pragma once
 
+#include <array>
```
Comment (calcmogul) [inline bms/src/BmsThread.h:3]: Add `#include <algorithm>` for std::transform(), and that should be everything.

**Pull request merged**